File: ratemyhike/src/main/java/com/example/ratemyhike/Service/UserService.java
package com.example.ratemyhike.Service;

import com.example.ratemyhike.Exceptions.UserWithIdDoesntExist;
import com.example.ratemyhike.Exceptions.UserWithUsernameDoesntExist;
import com.example.ratemyhike.Model.User;

import java.util.List;

public interface UserService {

    List<User> getAllUsers();

    User getUserById(long id) throws UserWithIdDoesntExist;

    User getUserByName(String name) throws UserWithUsernameDoesntExist;
}

File: Rate-My-Hike-FrontEnd/src/app/services/user.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from '../model/user';
import { AuthService } from './auth.service';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private httpClient: HttpClient, private authService:AuthService) { }
  token :any = this.authService.getToken();
  getAllUsers():Observable<User[]>{
    return this.httpClient.get<User[]>("http://localhost:9001/api/v1/users")
  }


  login(user:User):Observable<User>{
    return this.httpClient.post<User>("http://localhost:9001/api/v1/users/login", user)
  }
  getUserById(userid:number){
    return this.httpClient.get<User>("http://localhost:9001/api/v1/users/" + userid, {
      headers: new HttpHeaders().set('Authorization', `Bearer ${this.token}`)
    });
  }
  

}